FinancialTransactionsApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -118,5 +130,88 @@ public async Task UpdateAsync(Transaction transaction)
         {
             await _dynamoDbContext.SaveAsync(transaction.ToDatabase()).ConfigureAwait(false);
         }
+
+        public async Task<List<Transaction>> GetAllTransactionsForTheYearAsync(ExportTransactionQuery query)
+        {
+
+            var firstDay = new DateTime(DateTime.Now.Year, 1, 1);
+            var lastDay = new DateTime(DateTime.Now.Year, 12, 31);
+            string startDate = firstDay.ToString(AWSSDKUtils.ISO8601DateFormat);
+
+            string endDate = lastDay.ToString(AWSSDKUtils.ISO8601DateFormat);
+            QueryRequest queryRequest = new QueryRequest
+            {
+                TableName = "Transactions",
+                KeyConditionExpression = "pk = :V_pk",
+                FilterExpression = "target_id =:V_target_id AND transaction_date BETWEEN :V_startDate AND :V_endDate",
+                ExpressionAttributeValues = new Dictionary<string, AttributeValue>
+                    {
+                     {":V_pk", new AttributeValue {S = Pk}},
+                     {":V_target_id", new AttributeValue {S = query.TargetId.ToString()}},
+                     {":V_startDate", new AttributeValue { S = startDate }},
+                     {":V_endDate", new AttributeValue { S = endDate }}
+                    }
+            };
+
+            if (query.TransactionType != null)
+            {
+                queryRequest.FilterExpression += " AND transaction_type = :V_transaction_type";
+                queryRequest.ExpressionAttributeValues.Add(":V_transaction_type", new AttributeValue { S = query.TransactionType.ToString() });
+            }
+
+            var result = await _amazonDynamoDb.QueryAsync(queryRequest).ConfigureAwait(false);
+            var transactions = result.ToTransactions();
+            return transactions;
+        }
+
+        public async Task<List<Transaction>> GetAllTransactionRecordAsync(ExportTransactionQuery query)
```

Comment:
Could you please pass me the doc outlining the use case for this? How is this endpoint different to getting transactions for target ID between date X and Y ?

Reply:
This will be provided